apps/speed.c

```diff
@@ -2436,14 +2474,25 @@ int speed_main(int argc, char **argv)
             ret = 0;
             goto end;
         }
-        for (testnum = 0; testnum < size_num; testnum++) {
-            if (evp_cipher) {
 
-                names[D_EVP] = OBJ_nid2ln(EVP_CIPHER_nid(evp_cipher));
-                /*
-                 * -O3 -fschedule-insns messes up an optimization here!
-                 * names[D_EVP] somehow becomes NULL
-                 */
+        loopfunc = EVP_Update_loop;
+        if (evp_cipher != NULL) {
+            names[D_EVP] = OBJ_nid2ln(EVP_CIPHER_nid(evp_cipher));
+
+            if (EVP_CIPHER_mode(evp_cipher) == EVP_CIPH_CCM_MODE) {
+                loopfunc = EVP_Update_loop_ccm;
+            } else if (aead && (EVP_CIPHER_flags(evp_cipher) &
```

Comment:
is there a reason to use an cipher having an AEAD flag in another way than AEAD ?